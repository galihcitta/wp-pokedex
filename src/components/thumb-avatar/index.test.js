import React from "react";
import ThumbAvatar from "./index";
import StyleThumbAvatar from "./index.style";

let wrapper;

beforeEach(() => {
    wrapper = shallowWithTheme(
        <ThumbAvatar
            data-test="thumb-avatar-test-wrapper"
            type="circle"
            size="lg"
        />,
    );
});

describe("Header Menu Lang Component", () => {
    it("renders and create snapshot <ThumbAvatar />", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders success <ThumbAvatar />", () => {
        expect(wrapper.exists()).toBe(true);
    });

    it("renders and check avatar thumb image is ready <ThumbAvatar />", () => {
        expect(wrapper.find("img").prop("src")).toEqual("dummy-square.jpg");
    });

    it("renders and check avatar thumb have size <ThumbAvatar />", () => {
        expect(wrapper.prop("size")).toEqual("lg");
    });

    it("renders and check avatar thumb have radius <ThumbAvatar />", () => {
        expect(wrapper.prop("radius")).toEqual("circle");
    });

    it("check style thumb avatar is have size <ThumbAvatar />", () => {
        expect(
            mountWithTheme(
                <div>
                    <StyleThumbAvatar></StyleThumbAvatar>
                </div>,
            )
                .children()
                .prop("size"),
        ).toEqual("md");
    });
    it("check style thumb avatar is have radius <ThumbAvatar />", () => {
        expect(
            mountWithTheme(
                <div>
                    <StyleThumbAvatar></StyleThumbAvatar>
                </div>,
            )
                .children()
                .prop("radius"),
        ).toEqual("circle");
    });
});
