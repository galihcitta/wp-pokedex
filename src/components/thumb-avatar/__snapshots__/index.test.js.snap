// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Menu Lang Component renders and create snapshot <ThumbAvatar /> 1`] = `
<styled.div
  data-test="thumb-avatar-test-wrapper"
  image="dummy-square.jpg"
  radius="circle"
  size="lg"
  type="circle"
>
  <img
    alt=""
    className="image__pokemon"
    src="dummy-square.jpg"
  />
</styled.div>
`;
